javascript/word-count.js

```diff
@@ -0,0 +1,26 @@
+/**
+ * Created by justin maurer on 7/27/16.
+ */
+
+var Word = function() {};
+
+Word.prototype.count = function(phrase){
+	var re = /[\s\n]+/;
+	var phraseArray = phrase.trim().split(re);
+	var wordCounts = {};
+	for ( i=0; i < phraseArray.length; i++ ) {
+		var word = phraseArray[i].toLowerCase();
+		if ( wordCounts.hasOwnProperty(word) ) {
+			for( var key in wordCounts ) {
```

Comment:
You may be able to get rid of some of this nesting and reduce cognitive load for your fellow developer. Looping through each key in `wordCounts` may be unnecessary if you are checking for the existence of your `word` in `wordCounts`.